org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/RepositoryConfigurationManager.java

```diff
@@ -0,0 +1,144 @@
+import org.eclipse.winery.repository.backend.filebased.management.IRepositoryResolver;
+import org.eclipse.winery.repository.backend.filebased.management.RepositoryResolverFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RepositoryConfigurationManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RepositoryConfigurationManager.class);
+
+    private static File repositoriesConfiguration;
+    private static JsonNode repoNode = JsonNodeFactory.instance.objectNode();
+    private static Map<String, String> repositories = new HashMap<>();
+
+    private final MultiRepository multiRepository;
+
+    public RepositoryConfigurationManager(MultiRepository repository) {
+        this.multiRepository = repository;
+    }
+
+    public void initRepositoryConfigurationManager() throws IOException {
+        repositoriesConfiguration = new File(this.multiRepository.getRepositoryDep().toString(), Filename.FILENAME_JSON_REPOSITORIES);
+        readRepositoriesConfig();
+    }
+
+    protected void readRepositoriesConfig() throws IOException {
+        if (repositoriesConfiguration.exists()) {
+            LOGGER.info("Found Repositories file");
+            loadCfg(repositoriesConfiguration);
+            loadRepositories();
+        } else {
+            saveConfiguration(repositoriesConfiguration, repoNode);
+        }
+        LOGGER.info("Repositories loaded = " + this.multiRepository.getRepositoriesMap().keySet().size());
+        this.multiRepository.getRepositoriesMap().keySet().forEach(k -> LOGGER.info("Repository " + k.getRepositoryRoot() + " loaded."));
+    }
+
+    private void loadRepositories() {
+        Map<String, String> tempRepo = new HashMap<>(repositories);
+        reposByMap(tempRepo);
+        saveConfiguration(repositoriesConfiguration, repoNode);
+    }
+
+    private void saveConfiguration(File repoCfg, JsonNode repoNode) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+            objectMapper.writeValue(repoCfg, repoNode);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private Map<String, String> loadCfg(File cfg) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        byte[] jsonData = Files.readAllBytes(cfg.toPath());
+        JsonNode node = objectMapper.readTree(jsonData);
+
+        Map<String, String> result = new HashMap<>();
+        node.fieldNames().forEachRemaining(name -> {
+            String repoName = name.replace("https", "http");
```

Comment:
This assumes the repo name does not contain `https` except as a protocol. That's not necessarily true.